### PR TITLE
Write distinct message for error

### LINKED ISSUE #12: Print distinct message for check errors
Right now if a check errors out  we print the help text but we should print a different message to distinguish between a Check Failure and an error.  We can see say something as simple as, "This check encountered an error.  Please review the logs for more information."

### PR DESCRIPTION
**Pull request by edreyer-1**
Instead of the help text being displayed when there is an error with a check we want to display a message that directs the user to look at the logs to figure out the issue. We want there to be a distinction between a check failure and error.

Fixes #12

### CODE REVIEW

**Review by komish — COMMENTED** [review]
Thanks for taking care of this @edreyer-1!

Instead of adding `ErrorMessage` to our resultsText, what about repurposing our per-check Metadata and Help to fit the need? Happy to get more opinions on this.

Each check has a [Description](https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/main/certification/internal/shell/has_unique_tag.go#L50), [Suggestion](https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/main/certification/internal/shell/has_unique_tag.go#L59), and [Message](https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/main/certification/internal/shell/has_unique_tag.go#L59) which contains text indicating what the user should do in various scenarios. I'm seeing the Message and Description to be similar enough that we might not need both.

What do you think about capturing the gist of Message into the Description text, and then repurpose Message to answer the question of "where do I look when this check shows up in the error list"?

**Comment by edreyer-1** [comment]
@komish yes I agree with you that we could combine description and message as they seem to be very similar purposes. Seems Message often just repeats the description and says it doesn't meet that requirement. Though I think that is already indicated by the check being in the failed part of the results. 
I definitely agree with your suggestion

**Comment by samira-barouti** [comment]
I think we should probably involve the tool consumers to ensure the backward incompatible changes that we are making, is not breaking the pipeline. Other than that, I am also in favor of removing unnecessary data.

**Review by bcrochet — APPROVED** [review]
/lgtm

**Comment by komish** [comment]
> I think we should probably involve the tool consumers to ensure the backward incompatible changes that we are making, is not breaking the pipeline. Other than that, I am also in favor of removing unnecessary data.

@samira-barouti I don't think the suggested path causes backwards incompatible changes, given that it preserves the existing fields and doesn't add or remove any fields. In the suggested case, we just repurpose the existing fields so we don't have to make any changes to the schema.

**Comment by samira-barouti** [comment]
IMO this is still a backward incompatible change; it is true that the syntax does not change, but the semantic (i.e. behavior) of the API is changing.

**Comment by komish** [comment]
I'll send this PR over to those folks for review!

**Comment by komish** [comment]
Per conversation with @amisstea this shouldn't cause any issues on their end, so we should be good to change the meaning of `help` to guide the user in how to troubleshoot a check error. 

@edreyer-1 you should be set to implement!
@samira-barouti good catch!

File: certification/internal/shell/has_prohibited_packages.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func (p *HasNoProhibitedPackagesCheck) Name() string {
 }
 func (p *HasNoProhibitedPackagesCheck) Metadata() certification.Metadata {
 	return certification.Metadata{
-		Description:      "Checks to ensure that the image in use does not contain prohibited packages.",
+		Description:      "Checks to ensure that the image in use does not include Red Hat Enterprise Linux (RHEL) kernel packages.",
```
Comment (komish) [inline certification/internal/shell/has_prohibited_packages.go:65]: I _think_ the packages that we don't want to see include more than just kernel-related packages. Can you instead say something like `does not include prohibited packages, such as ...`

File: certification/internal/shell/has_required_labels.go [hunk 2]
```diff
@@ -59,7 +59,7 @@ func (p *HasRequiredLabelsCheck) Metadata() certification.Metadata {
 
 func (p *HasRequiredLabelsCheck) Help() certification.HelpText {
 	return certification.HelpText{
-		Message:    "Container images must include the following metadata: name, vendor, version, release, summary, description",
+		Message:    "Check Check HasRequiredLabel encountered an error. Please review the preflight.log file for more information. encountered an error. Please review the preflight.log file for more information.",
```
Comment (komish) [inline certification/internal/shell/has_required_labels.go:62]: This looks like a duplicate paste

**Review by komish — CHANGES_REQUESTED** [review]
Just a couple of updates!

File: certification/internal/shell/has_required_labels.go (hunk 2 above)
Comment (edreyer-1) [reply certification/internal/shell/has_required_labels.go:62]: oops looks like I did double paste good catch

**Review by komish — APPROVED** [review]
/lgtm

Thanks @edreyer-1!

**Pull request merged**